src/current/v25.1/do.md

```diff
@@ -0,0 +1,101 @@
+---
+title: DO
+summary: The DO statement executes a SQL or PL/pgSQL code block.
+toc: true
+keywords:
+docs_area: reference.sql
+---
+
+The `DO` [statement]({% link {{ page.version.version }}/sql-statements.md %}) defines a code block that executes SQL or [PL/pgSQL]({% link {{ page.version.version }}/plpgsql.md %}) syntax.
+
+## Required privileges
+
+- To execute a `DO` block, a user must have at least the [`USAGE` privilege]({% link {{ page.version.version }}/security-reference/authorization.md %}#supported-privileges) on the schema where the `DO` block is being executed.
+- To define a `DO` block with a [user-defined type]({% link {{ page.version.version }}/create-type.md %}), a user must have `USAGE` privilege on the user-defined type.
+
+## Synopsis
+
+<div>
+{% remote_include https://raw.githubusercontent.com/cockroachdb/generated-diagrams/{{ page.release_info.crdb_branch_name }}/grammar_svg/do.html %}
+</div>
+
+## Parameters
```

Comment:
Note there's also an optional `LANGUAGE foo`. SQL is disallowed and we currently only support PL/pgSQL out of the options Postgres allows.

Reply:
Yep, this is in the diagram. But I didn't know about SQL being disallowed. So the diagram should only show an optional `LANGUAGE PLPGSQL`?

Reply:
>So the diagram should only show an optional LANGUAGE PLPGSQL?

Yeah, that sounds good.